hazelcast/src/main/java/com/hazelcast/map/impl/operation/EvictBatchBackupOperation.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2008-2017, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.map.impl.operation;
+
+import com.hazelcast.internal.cluster.Versions;
+import com.hazelcast.map.impl.MapDataSerializerHook;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.spi.BackupOperation;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.LinkedList;
+
+import static com.hazelcast.util.CollectionUtil.isNotEmpty;
+
+public class EvictBatchBackupOperation extends MapOperation implements BackupOperation {
+
+    private String name;
+    private Collection<Data> expiredKeys;
+    private int ownerPartitionEntryCount;
+
+    public EvictBatchBackupOperation() {
+    }
+
+    public EvictBatchBackupOperation(String name, Collection<Data> expiredKeys, int ownerPartitionEntryCount) {
+        super(name);
+
+        assert isNotEmpty(expiredKeys);
+        assert name != null;
+
+        this.name = name;
+        this.expiredKeys = expiredKeys;
+        this.ownerPartitionEntryCount = ownerPartitionEntryCount;
+    }
+
+    @Override
+    public void run() {
+        assert getNodeEngine().getClusterService().getClusterVersion().isGreaterOrEqual(Versions.V3_9);
+
+        for (Data key : expiredKeys) {
+            recordStore.evict(key, true);
+        }
+
+        // equalize backup entry count to owner entry count to have identical memory occupancy
+        int diff = recordStore.size() - ownerPartitionEntryCount;
```

Comment:
please see [this](https://hazelcast.atlassian.net/wiki/display/EN/Partition+Owner+Should+Be+Responsible+for+Expirations+and+Evictions+on+Replicas+-+Design?focusedCommentId=131339648#comment-131339648)